Rick-and-Morty/Rick And Morty/Views/CharacterListView.swift

```diff
@@ -5,12 +5,21 @@ struct CharacterListView: View {
     
     var body: some View {
         NavigationView {
-            List(viewModel.characterListViewState.characters, id: \.id) { character in
-                CharacterCard(viewModel: CharacterCardViewModel(character: character))
-                    .onAppear(perform: {
-                        viewModel.loadIfNeeded(characterID: character.id)
-                    })
+            ZStack {
+                List(viewModel.characterListViewState.characters, id: \.id) { character in
+                    CharacterCard(viewModel: CharacterCardViewModel(character: character))
+                        .onAppear(perform: {
+                            viewModel.loadIfNeeded(characterID: character.id)
+                        })
+                }
+                
+                if viewModel.characterListViewState.state == .error {
+                    ErrorView(errorMessage: viewModel.characterListViewState.errorMessage, refreshAction: viewModel.loadCharacters)
+                } else if viewModel.characterListViewState.state == .loading {
+                    ProgressView()
+                }
```

Comment:
this logic should be in the VM. It shouldnt be possible to unit test the view :)